### ISSUE TITLE
Change FinancialAidSerializer to use a ModelSerializer

### ISSUE THREAD

**Issue by noisecapella** [mitodl/micromasters#1063]
From #1053

**Comment by pdpinch** [comment]
Is this a refactoring, or does it affect critical functionality?

**Comment by noisecapella** [comment]
It's a refactoring issue. Switching to a ModelSerializer would make it much more flexible if we needed to expand functionality, but what is there now fits the immediate use case for financial aid.

**Comment by alvinsiu** [comment]
I'm not sure switching to a ModelSerializer would be too helpful in this case.  According to the docs:

> It's important to remember that ModelSerializer classes don't do anything particularly magical, they are simply a shortcut for creating serializer classes:
>
> - An automatically determined set of fields.
> - Simple default implementations for the create() and update() methods.

We initially tried to make it a ModelSerializer for the purpose of using the built-in validation for fields, but because our view required a non-standard set of POST data, one of which (`program_id`) isn't on the FinancialAid model, a lot of the built-in validation actually prevents us from using this field.

I'm happy to discuss further and am open to be pursuaded otherwise, but I don't feel this change necessarily warrants an issue, since it doesn't provide any immediate benefit and if this change is needed, it would probably be done as part of a different issue.

**Comment by noisecapella** [comment]
You're right that `program_id` needs some special handling, but I think that can be done inside `ModelSerializer.validate` which can return an altered set of data for this purpose. The other two fields can be set as fields in the `Meta` class. I haven't done any investigating on this but we should leave this issue open so we can explore this more.

**Comment by alvinsiu** [comment]
We tried returning an altered set of data in `ModelSerializer.validate`, but there were issues with `ModelSerializer` doing automatic type checking and some other things I can't remember that made it not work as expected (I think it was during `ModelSerializer.to_representation`).  In any case, it felt a lot like we were trying to wrangle `ModelSerializer` into doing something it didn't want to do.

Let's leave this issue open anyway for further exploration like you suggested.
